articulated_rover:
  # Publish all joint states -----------------------------------
  # Creates the /joint_states topic necessary in ROS
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50


  # Wheel Velocity Controllers ---------------------------------------
  left_front_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_front_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}

  right_front_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_front_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}



  left_middle_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_middle_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}

  right_middle_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_middle_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}



  left_rear_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_rear_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}

  right_rear_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_rear_wheel_joint
    pid: {p: 10.0, i: 0.01, d: 0.1}


  # Wheel Trees Position Controllers ---------------------------------------


  # Wheel Steering Position Controllers ---------------------------------------


  # Sensor Mast Position Controllers ---------------------------------------




  # Head Position Controllers ---------------------------------------
  head_yaw_position_controller:
    type: position_controllers/JointPositionController
    joint: head_yaw_joint
    pid: {p: 30.0, i: 0.05, d: 0.1}

  head_pitch_position_controller:
    type: position_controllers/JointPositionController
    joint: head_pitch_joint
    pid: {p: 30.0, i: 0.05, d: 0.1}




  # Arm Position Controllers ---------------------------------------
  arm_shoulder_yaw_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_shoulder_yaw_joint
    pid: {p: 50.0, i: 0.1, d: 0.5}

  arm_shoulder_pitch_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_shoulder_pitch_joint
    pid: {p: 50.0, i: 0.1, d: 0.5}

  arm_elbow_pitch_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_elbow_pitch_joint
    pid: {p: 40.0, i: 0.1, d: 0.4}

  arm_wrist_pitch_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_wrist_pitch_joint
    pid: {p: 30.0, i: 0.05, d: 0.3}

  arm_wrist_yaw_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_wrist_yaw_joint
    pid: {p: 30.0, i: 0.05, d: 0.3}



  arm_tool_roll_position_controller:
    type: position_controllers/JointPositionController
    joint: arm_tool_roll_joint
    pid: {p: 20.0, i: 0.02, d: 0.2}